examples/src/armory/examples/object_detection/visdrone_yolov5_robustdpatch.py

```diff
@@ -0,0 +1,155 @@
+import armory.model.object_detection
+
+logger = logging.getLogger(__name__)
+
+
+class RobustDPatchModule(LightningModule):
+
+    def __init__(self, model: armory.model.object_detection.YoloV5ObjectDetector):
+        super().__init__()
+        self.model = model
+        self.target_spec = armory.data.TorchBoundingBoxSpec(
+            format=armory.data.BBoxFormat.CXCYWH
+        )
+        self.patch_shape = (3, 50, 50)
+        self.patch_location = (295, 295)  # middle of 640x640
+        # TODO non-zero min value
+        self.patch_min = 0
+        self.patch_max = self.model.inputs_spec.scale.max
+        self.patch = torch.randint(0, 255, self.patch_shape) / 255 * self.patch_max
+        self.patch.requires_grad = True
+        self.initial_patch = self.patch.clone()
+        self.targeted = False
+        self.learning_rate = 0.1
+        self.augmentation = kornia.augmentation.container.ImageSequential(
+            kornia.augmentation.RandomHorizontalFlip(p=0.5),
+            kornia.augmentation.RandomBrightness(brightness=(0.75, 1.25), p=0.5),
+            kornia.augmentation.RandomRotation(degrees=15, p=0.5),
+            random_apply=True,
+        )
+
+    def configure_optimizers(self):
+        return torch.optim.SGD([self.patch], lr=self.learning_rate, momentum=0.9)
+
+    def on_train_epoch_end(self):
+        if isinstance(self.logger, MLFlowLogger):
+            if self.current_epoch % 5 == 0:
+                patch_np = (
+                    self.patch.detach().cpu().numpy().transpose(1, 2, 0) * 255
+                ).astype("uint8")
+                patch = PIL.Image.fromarray(patch_np)
+                self.logger.experiment.log_image(
+                    self.logger.run_id, patch, f"patch_epoch_{self.current_epoch}.png"
+                )
+
+    def training_step(self, batch: armory.data.Batch, batch_idx: int):
+        # Get inputs as Tensor
+        inputs = batch.inputs.get(self.model.inputs_spec)
+        assert isinstance(inputs, torch.Tensor)
+
+        # Apply patch to image
+        x_1 = random.randint(0, inputs.shape[3] - self.patch_shape[2])
+        y_1 = random.randint(0, inputs.shape[2] - self.patch_shape[1])
+        # x_1, y_1 = self.patch_location
+        x_2 = x_1 + self.patch_shape[1]
+        y_2 = y_1 + self.patch_shape[2]
+        inputs_with_patch = inputs.clone()
+        inputs_with_patch[:, :, x_1:x_2, y_1:y_2] = self.patch
+
+        # Apply random augmentations to images
+        inputs_with_augmentations = self.augmentation(inputs_with_patch)
+
+        # Get targets as Tensor
+        _, _, height, width = inputs.shape
+        targets = batch.targets.get(self.target_spec)
+        yolo_targets = self._to_yolo_targets(targets, height, width, self.model.device)
+
+        # Get loss from model outputs
+        self.model.train()
+        loss_components = self.model(inputs_with_augmentations, yolo_targets)
+        loss = loss_components["loss_total"]
```

Comment:
Robust DPatch optimizes the patch by increasing the loss, equivalently minimizing the negative loss.
```
loss = - loss_components["loss_total"]
```